### DISCUSSION TITLE
Can this be used just to show diagrams?

### DISCUSSION THREAD

**Discussion by ppazos** [eckig/graph-editor#194] [Q&A]
I need to build diagrams with connected boxes dynamically from input data, but don't need the edit functionality. Though I would need to draw boxes that automatically distribute themselves in the canvas to avoid overlaps. Can this be done with this library? Thanks!

**Comment by eckig** [comment]
- Editable state can be configured.
- Auto-Layout not so much.. i made good progress integrating the [eclipse layout kernel](https://github.com/eclipse/elk) but never got it over the finish line.

**Comment by ppazos** [comment]
Thanks @eckig 

About the layout, you might be able to get a force-directed layout from this project https://github.com/brunomnsilva/JavaFXSmartGraph which looks really nice!
